### PR TITLE
update PSI-MS.obo 

### PR DESCRIPTION
**Pull request by timosachsenberg**
___

### **PR Type**
enhancement, tests

___

### **Description**
- Updated the version of the `Proteomics Standards Initiative Mass Spectrometry Ontology` from "4.1.56" to "4.1.155" in multiple test and source files.
- Ensured consistency across various test data files by updating the version number.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>MzTabM_test.cpp</strong><dd><code>Update version in MzTabM_test.cpp</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/tests/class_tests/openms/source/MzTabM_test.cpp
- Updated `meta_cv.version` from "4.1.56" to "4.1.155".

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7506/files#diff-1468cbc044b60e4daea2dedd5736e090c09ba411e11c67204c20dc8401e65e7c">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>MzTabMFile_output_1.mztab</strong><dd><code>Update version in MzTabMFile_output_1.mztab</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/tests/class_tests/openms/data/MzTabMFile_output_1.mztab
- Updated `MTD.cv[1]-version` from "4.1.56" to "4.1.155".

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7506/files#diff-7944e77868350a5f72d46f77276beda2f500eaf9c94e40e077c255f032d90071">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>AccurateMassSearchEngine_output1_mztabm_featureXML.mzTab</strong><dd><code>Update version in </code><br><code>AccurateMassSearchEngine_output1_mztabm_featureXML.mzTab</code></dd></summary>
<hr>

src/tests/class_tests/openms/data/AccurateMassSearchEngine_output1_mztabm_featureXML.mzTab
- Updated `MTD.cv[1]-version` from "4.1.56" to "4.1.155".

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7506/files#diff-8cd916885b8d76b13779f71a8a5e5bcd57eeee3ea45ba54b5ada63a2b361983e">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>AccurateMassSearchEngine_output2_mztabm_featureXML.mzTab</strong><dd><code>Update version in </code><br><code>AccurateMassSearchEngine_output2_mztabm_featureXML.mzTab</code></dd></summary>
<hr>

src/tests/class_tests/openms/data/AccurateMassSearchEngine_output2_mztabm_featureXML.mzTab
- Updated `MTD.cv[1]-version` from "4.1.56" to "4.1.155".

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7506/files#diff-72be706cce978b253348f445ed2654452a7d47aa1b2af34410f8f9ac716e3cbf">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>QCCalculator_2_output.mzQC</strong><dd><code>Update version in QCCalculator_2_output.mzQC</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/tests/topp/QCCalculator_2_output.mzQC
<li>Updated version in the <code>Proteomics Standards Initiative Mass </code><br><code>Spectrometry Ontology</code> from "4.1.56" to "4.1.155".<br>

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7506/files#diff-a7582dc5573480c10cbacd680b7ed04eed7c0a5fe9f9df492fcd34cd977bfd3c">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>MzQCFile.cpp</strong><dd><code>Update version in MzQCFile.cpp</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/source/FORMAT/MzQCFile.cpp
<li>Updated version in the <code>Proteomics Standards Initiative Mass </code><br><code>Spectrometry Ontology</code> from "4.1.56" to "4.1.155".<br>

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7506/files#diff-1503bf86e49c0565193abe9a26d4063c6b25b4e39baf9dcb580139346858cec1">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by timosachsenberg** [comment]
don't merge yet

**Comment by timosachsenberg** [comment]
@cbielow 
I get a lot of errors after the update:
```
153: Error - Value of CV term not allowed: 'MS:1000569 - SHA-1' value='81be39fb2700ab2f3c8b2234b91274968b6899b1' at element '/mzML/fileDescription/sourceFileList/sourceFile'
153: Error - Value of CV term not allowed: 'MS:1000586 - contact name' value='Pennington, William' at element '/mzML/fileDescription/contact'
153: Error - Value of CV term not allowed: 'MS:1000590 - contact affiliation' value='dort' at element '/mzML/fileDescription/contact'
153: Error - Value of CV term not allowed: 'MS:1000587 - contact address' value='Higglesworth University, 12 Higglesworth Avenue, 12045, HI, USA' at element '/mzML/fileDescription/contact'
153: Error - Value of CV term not allowed: 'MS:1000588 - contact URL' value='http://www.higglesworth.edu/' at element '/mzML/fileDescription/contact'
153: Error - Value of CV term not allowed: 'MS:1000589 - contact email' value='wpennington@higglesworth.edu' at element '/mzML/fileDescription/contact'
153: Error - Value of CV term not allowed: 'MS:1000586 - contact name' value='Threepwood, Guybrush' at element '/mzML/fileDescription/contact'
153: Error - Value of CV term not allowed: 'MS:1000590 - contact affiliation' value='Monkey Island' at element '/mzML/fileDescription/contact'
153: Error - Value of CV term not allowed: 'MS:1000001 - sample number' value='5' at element '/mzML/sampleList/sample'
153: Error - Value of CV term not allowed: 'MS:1000004 - sample mass' value='11.7' at element '/mzML/sampleList/sample'
2061: Warning: while loading term 'MS:1003303' of CV 'PSI-MS': parent term name 'single identification attribute' and id 'MS:1002694' differ.
2061: Warning: while loading term 'MS:1003337' of CV 'PSI-MS': parent term name 'interaction score derived from  crosslinking' and id 'MS:1002664' differ.
2061: Warning: while loading term 'MS:1003338' of CV 'PSI-MS': parent term name 'interaction score derived from  crosslinking' and id 'MS:1002664' differ.
2061: Warning: while loading term 'MS:1003408' of CV 'PSI-MS': parent term name 'PSM-level search engine specific statistic' and id 'MS:1003407' differ.
2061: Error: Unexpected internal error (the value 'cross-linking search' was used but is not valid; Invalid CV name!)
```
any idea why this is the case?
in the obo we have e.g.
```
[Term]
id: MS:1000590
name: contact affiliation
def: "Home institution of the contact person." [PSI:MS]
is_a: MS:1000585 ! contact attribute
relationship: has_value_type xsd:string ! The allowed value-type for this CV term
```

**Comment by timosachsenberg** [comment]
change between old version
removed: 
```
xref: value-type:xsd\:string "The allowed value-type for this CV term."
```
added:
```
relationship: has_value_type xsd:string ! The allowed value-type for this CV term
```

**Comment by timosachsenberg** [comment]
hmm was able to fix all bugs except: https://cdash.seqan.de/viewTest.php?onlyfailed&buildid=269008
currently a bit lost why indexList parsing in the DOM parser doesn't work anymore. Should not really depend on the obo stuff... 

the test files seem to contain the indexList element: https://raw.githubusercontent.com/OpenMS/OpenMS/develop/src/tests/topp/GNPSExport_mzml1.mzML

the error output says
```
IndexedMzMLDecoder::domParseIndexedEnd Error: no indexList element found:

<indexedmzML><index name="spectrum">
      <offset idRef="controllerType=0 controllerNumber=1 scan=621">5024</offset>
      <offset idRef="controllerType=0 controllerNumber=1 scan=622">12588</offset>
```
but the mzML contains:
```
<indexedmzML>
...
  <indexList count="2">
    <index name="spectrum">
      <offset idRef="controllerType=0 controllerNumber=1 scan=621">5024</offset>
      <offset idRef="controllerType=0 controllerNumber=1 scan=622">12588</offset>
```
@cbielow or @jpfeuffer any clue how this could happen???

File: src/openms/source/FORMAT/HANDLERS/MzIdentMLHandler.cpp [hunk 1]
```diff
@@ -971,7 +971,7 @@ namespace OpenMS::Internal
       //--------------------------------------------------------------------------------------------
       os << "<cvList>\n"
          << "\t<cv id=\"PSI-MS\" fullName=\"Proteomics Standards Initiative Mass Spectrometry Vocabularies\" "
-         << "uri=\"https://raw.githubusercontent.com/HUPO-PSI/psi-ms-CV/master/psi-ms.obo\" "
+         << "uri=\"http://purl.obolibrary.org/obo/ms/psi-ms.obo\" "
```
Comment (jpfeuffer) [inline src/openms/source/FORMAT/HANDLERS/MzIdentMLHandler.cpp:974]: There is really no secure connection to that obo? That's really sad..

File: src/openms/source/FORMAT/ControlledVocabulary.cpp [hunk 2]
```diff
@@ -183,8 +183,22 @@ namespace OpenMS
       }
       if (line_wo_spaces.hasPrefix("remark:URL:"))
       {
-        // https://
-        url_ = line.substr(line.find_first_of('/') - 7).trim();
+        // Find the position of "http://" or "https://"
+        size_t httpPos = line.find("http://");
+        size_t httpsPos = line.find("https://");
```
Comment (jpfeuffer) [inline src/openms/source/FORMAT/ControlledVocabulary.cpp:188]: I would do the "if end" here already but it's ok.

File: src/openms/source/FORMAT/ControlledVocabulary.cpp [hunk 3]
```diff
@@ -382,7 +396,8 @@ namespace OpenMS
             term.xref_type = CVTerm::XSD_STRING;
             continue;
           }
-          if (line_wo_spaces.hasSubstring("xsd:integer") || line_wo_spaces.hasSubstring("value-type:xsd:int"))
+          if (line_wo_spaces.hasSubstring("xsd:integer") 
+          || line_wo_spaces.hasSubstring("xsd:int"))
```
Comment (jpfeuffer) [inline src/openms/source/FORMAT/ControlledVocabulary.cpp:400]: Why the change here? I agree, yours looks more correct but maybe this really somehow has to be valuetype:xsd:int

File: src/openms/source/FORMAT/HANDLERS/MzIdentMLHandler.cpp (hunk 1 above)
Comment (timosachsenberg) [reply src/openms/source/FORMAT/HANDLERS/MzIdentMLHandler.cpp:974]: Yeah annoying but I made it now work with http as well.

File: src/openms/source/FORMAT/ControlledVocabulary.cpp (hunk 3 above)
Comment (timosachsenberg) [reply src/openms/source/FORMAT/ControlledVocabulary.cpp:400]: This was a copy and paste bug that I introduced before in the previous PR (support for relationship has_value_type inner obo) and now is fixed.

**Comment by timosachsenberg** [comment]
the mzML index got messed up in those files. fixed in #7510

**Pull request closed without merging**